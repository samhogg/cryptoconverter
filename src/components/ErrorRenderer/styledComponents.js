import styled from "styled-components";

const ErrorWrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  opacity: 0.95;
  background-color: ${({ theme }) => theme.colors.error};
  padding: 3rem 0.5rem;
  color: white;
  text-align: center;
  z-index: 1;
`;

const RetryButton = styled.button`
  display: block;
  margin: 0 auto;
  margin-top: 0.5rem;
  border-radius: 3px;
  cursor: pointer;
`;

export { ErrorWrapper, RetryButton };
