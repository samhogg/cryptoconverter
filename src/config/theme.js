const theme = {
  colors: {
    primary: "#0062AE",
    error: "#be0000"
  },

  // System Font Stack
  // Adapted from https://github.com/mrmrs/css-system-fonts
  fontStack: `-apple-system, BlinkMacSystemFont,
    /* MacOS and iOS */ "avenir next", avenir, /* MacOS and iOS */ "Segoe UI",
    /* Windows */ "lucida grande", /* Older MacOS */ "helvetica neue", helvetica,
    /* Older MacOS */ "Fira Sans", /* Firefox OS */ roboto, noto,
    /* Google stuff */ "Droid Sans", /* Old Google stuff */ cantarell, oxygen,
    ubuntu, /* Linux stuff */ "franklin gothic medium", "century gothic",
    /* Windows stuff */ "Liberation Sans", /* Linux */ sans-serif; /* Everything else */`
};

export default theme;
