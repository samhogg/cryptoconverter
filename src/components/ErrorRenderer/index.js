import React from "react";
import PropTypes from "prop-types";
import { ErrorWrapper, RetryButton } from "./styledComponents";

const ErrorRenderer = ({ error, retry }) => (
  <ErrorWrapper>
    {error}
    <RetryButton onClick={retry}>Retry</RetryButton>
  </ErrorWrapper>
);

ErrorRenderer.propTypes = {
  error: PropTypes.string.isRequired,
  retry: PropTypes.func.isRequired
};

export default ErrorRenderer;
