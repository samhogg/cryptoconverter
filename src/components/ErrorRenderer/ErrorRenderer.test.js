import React from "react";
import renderer from "react-test-renderer";
import ErrorRenderer from ".";
import { ThemeProvider } from "styled-components";
import theme from "../../config/theme";

it("Renders an error", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ErrorRenderer retry={() => {}} error="Error Message" />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
