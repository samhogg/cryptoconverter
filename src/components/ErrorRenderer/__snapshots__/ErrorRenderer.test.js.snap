// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders an error 1`] = `
<div
  className="styledComponents__ErrorWrapper-bxVaWE iTsRMw"
>
  Error Message
  <button
    className="styledComponents__RetryButton-eBsaHs iRHKIs"
    onClick={[Function]}
  >
    Retry
  </button>
</div>
`;
